src/ChunkMap.cpp

```diff
@@ -1088,7 +1088,7 @@ int  cChunkMap::GetHeight(int a_BlockX, int a_BlockZ)
 		}
 
 		// The chunk is not valid, wait for it to become valid:
-		cCSUnlock Unlock(Lock);
+		m_CSLayers.Unlock();
 		m_evtChunkValid.Wait();
```

Comment:
This is  weird bit of code, because what the original bit of code was doing was locking when Unlock was released, then immediately unlocked when Lock is released. I suspect the real answer is to not use cCSLock here.